homeassistant/components/modbus/base_platform.py

```diff
@@ -185,6 +187,8 @@ def unpack_structure_result(self, registers: list[int]) -> str | None:
             v_result = []
             for entry in val:
                 v_temp = self._scale * entry + self._offset
+                if self._min_value and abs(v_temp) < self._min_value:
```

Comment:
This will not work if the user configured min_value = 0 (which is the default), nor will it work if it is the temperature in my frigde (-6).

Reply:
If min_value is 0 (the default) this will have no effect.
If the fridge temp is -3 and the min is 4 then --> 0
If the fridge temp is -4 and the min is 4 then --> -4
This is due to some sensors having noise around zero.
Would you prefer a smallest_positive_value and smallest_negative_value?
If you want +/- smallest control then you would need both

Reply:
in case of the fridge min would be something like -10 and max would be 0.

It does not make sense that if fridge temp is -4 and min is 4 --> -4, this should be 4 because 4 > -4. Min value is the minimum value. 

My preference is simple value < min_value --> min_value
value > max_value --> max_value.

Reply:
you need both, because we handle negative and positive values. You could redefine it as abs_min_value, but I think most users will be confused about how to use it.

Many sensors are only defined for a specific range, and outside that range the value cannot be trusted, that is the general case for a min/max value.

Reply:
ok ... lets worry take your fridge and say temp is not reliable around 0 (I am using energy device but I thought about temp)
I could not use min_value or max_value to help this
Maybe call it something different as its the small_value_threshold?
Not sure what else to call it as min/max as you put it will not have the same effect of small values --> 0

Reply:
Maybe we need all 3 .. happy to add 3 if you prefer ... min_value, max_value, and small_value_limit (or some such) ?

Reply:
No I really do not see the need for small_value_limit, that is perfectly covered with min/max, which are the traditional limiters.

Reply:
mmm ... so take my example of the energy unit:
I would need a MIN_THRESHOLD and a MIN_VALUE
i.e. I trat anything below min_value: 0.0251 as 0
so I would need to specify:
min_value: 0
min_threshold: 0.0251
Would you prefer min and max pairs like that?

Reply:
Pushed an update that will do this

Reply:
I think that meets what you are trying to so, but is flexible enough to cover my energy device as well, and in the normal case min_value and max_value work exactly as you describe ... but you can override the thresholds for these if you want...
If you are happy, please resolve and I will document